src/message/mailbox/types.rs

```diff
@@ -327,6 +327,87 @@ impl FromStr for Mailboxes {
     }
 }
 
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.6
+fn write_word(f: &mut Formatter<'_>, s: &str) -> FmtResult {
```

Comment:
This function is potentially reusable for any header that requires a `phrase` or sequence of `word`s. Let me know if there is a better home for it. it would even be useful for people who are implementing their own headers.

Reply:
Resolving based on https://github.com/lettre/lettre/pull/700#issuecomment-968881288. This should eventually live in the [email-encoding crate](https://github.com/lettre/email-encoding) but we will merge for now to fix lettre and we can migrate this to the new crate once the dust settles on the migration.